### ISSUE TITLE
Using default/blank save path on Linux prepends and appends garbage to save filenames (".xdp-" ".sav-mfHSlR") if game is on non-secondary drive

### ISSUE THREAD

**Issue by NanashiEwSa** [melonDS-emu/melonDS#1621]
Machine is running updated Linux Mint, MelonDS 9.5 was Installed from Flathub via Mint's software manager. 

Issue: Save files (both battery and savestate) created by MelonDS all have ".xdp" in front of them and have garbage after the filetype (.sav) like "sav-mfHSlR", "sav-BSS9i6", "sav-flQjtI", or "sav-Oix2g9" (generating file names like ".xdp-[gamename].sav-mfHSlR"). These files load normally while existing .sav files with garbage free names in the directory are not recognized. Upon system reboot the garbage that gets appended to a file changes, and it only reads files with that garbage in the filename (not the previous boot's garbage appendix). Removing this garbage from the filename allows it to be imported or used (if a save file directory is set and it's put there) without further alteration.  Can reproduce issue with multiple games and multiple drives.

Workaround: Set a path for save files and locate any existing .sav files there (with garbage stripped if needed). Alternatively, locate game files on system drive (which is, of course, suboptimal since if you're putting game files on an alternate drive you're clearly trying to avoid using system drive space).

**Comment by patataofcourse** [comment]
IDK if this is something Flathub related, but I'm running Manjaro and running melonDS from the AUR and that doesn't happen to me. Could you try a build from Actions and see if the issue persists?

**Comment by draskmont** [comment]
This issues is old, but just in case anyone have this problem. This can be solved by using Flatseal or the command line to add the directory that contains the roms/saves as a flatpak filesystem override.

**Comment by outusuke** [comment]
It would be helpful if the default save folder were set to a directory within /home by default. This would simplify the configuration process for users and avoid the need for manual permission adjustments in Flatpak environments

**Comment by nadiaholmquist** [comment]
I agree with you on that, though it might be too surprising of a change of behavior to change it now, since all prior versions put the save file next to the ROM by default. I would have to take it up with the other contributors. But if it were to be changed, I could set better flatpak sandboxing permissions on the package which would be nice.
